quill-jdbc/src/main/scala/io/getquill/context/jdbc/Decoders.scala

```diff
@@ -55,7 +55,7 @@ trait Decoders {
   implicit val doubleDecoder: Decoder[Double]         = decoder(_.getDouble)
   implicit val byteArrayDecoder: Decoder[Array[Byte]] = decoder(_.getBytes)
   implicit val dateDecoder: Decoder[util.Date] =
-    decoder((index, row, session) => new util.Date(row.getTimestamp(index, Calendar.getInstance(dateTimeZone)).getTime))
```

Comment:
Are we sure this Calendar instance is thread-safe?

Reply:
It isn't thread-safe. I don't see problems with it, because java.util.Date doesn't mutate calendar, just reads timezone.

Reply:
@juliano What do you think? 🤔

Reply:
It seems reasonable to me, looking at the postgres driver implementation, the calendar is just a way to read the timezone. Can you see any possible problems?

Reply:
>  Can you see any possible problems?

I don't know enough the code base nor how this work to know any better

Reply:
I think it'd be safer to stay with the current code tho as we don't know what the driver are doing with this Calendar instance. I don't that's expensive anyway

See:
<img width="943" alt="image" src="https://github.com/zio/zio-quill/assets/1193670/ec328af5-939d-4818-9fc2-a27b419cf0dc">

Found here: https://www.tabnine.com/code/java/methods/java.sql.PreparedStatement/setTimestamp

Reply:
You are right, being a possibility we should be extra careful, even if we have a gain changing the code

Reply:
Ok though, let's close PR and issue. 
I've found it while optimizing cpu usage of my app - that decoder takes ~3% process time. I'll think about workaround in client code.

Reply:
@gr1ev0us we really appreciate your contribution here. Maybe we are being too cautious about something that will indeed optimize the cpu usage, but the uncertainty of how that could bring other problems (we don't know how the db drivers will use the calendar instance) is the reason of that. I hope it makes sense to you.

Reply:
Maybe cloning the Calendar would be cheaper in terms of CPU?
Do we have an idea why Instantiating a Calendar takes that much CPU?